src/python/WMCore/JobStateMachine/ChangeState.py

```diff
@@ -318,7 +318,10 @@ def recordInCouch(self, jobs, newstate, oldstate, updatesummary = False):
                     jobState = "jobfailed"
                 else:
                     jobState = newstate
-                    
+                
+                # there is race condition updating couch record location and job is completed0
+                # for the fast fail job, it could miss the location update
+                job["location"] = job["fwjr"].getSiteName() or job.get("location", "UnKnown")     
```

Comment:
This will give the fast fail job (missed by JobTracker) correct site name if fwjr is created. So we might not need to look at schedd log to track those jobs although we can't have correct information if fwjr is not generated